packages/frontend-api/src/Model/Resolver/Products/ProductResolverMap.php

```diff
@@ -4,190 +4,93 @@
 
 namespace Shopsys\FrontendApiBundle\Model\Resolver\Products;
 
+use ArrayObject;
 use BadMethodCallException;
+use GraphQL\Type\Definition\ResolveInfo;
+use Overblog\GraphQLBundle\Definition\ArgumentInterface;
+use Overblog\GraphQLBundle\Resolver\FieldResolver;
 use Overblog\GraphQLBundle\Resolver\ResolverMap;
 use Shopsys\FrameworkBundle\Component\Domain\Domain;
 use Shopsys\FrameworkBundle\Model\Category\CategoryFacade;
-use Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser;
-use Shopsys\FrameworkBundle\Model\Product\Brand\BrandFacade;
 use Shopsys\FrameworkBundle\Model\Product\Collection\ProductCollectionFacade;
 use Shopsys\FrameworkBundle\Model\Product\Flag\FlagFacade;
 use Shopsys\FrameworkBundle\Model\Product\Product;
-use Shopsys\FrameworkBundle\Model\Product\ProductFacade;
-use Shopsys\FrontendApiBundle\Model\Parameter\ParameterWithValuesFactory;
-use Shopsys\FrontendApiBundle\Model\Product\ProductAccessoryFacade;
+use Shopsys\FrontendApiBundle\Model\Resolver\Products\DataMapper\ProductArrayFieldMapper;
+use Shopsys\FrontendApiBundle\Model\Resolver\Products\DataMapper\ProductEntityFieldMapper;
 
 class ProductResolverMap extends ResolverMap
```

Comment:
Why is not the whole class deprecated and new class `ProductResolverByUuidMap` introduced? Would not make it more sence, not just because of tons of changes here?

Reply:
That's not so easy... This map is not used just for ByUuid resolver, but for any work with the product in FE API (promoted product, lists), so the name is not ideal.

Changes are not so terrifying as it may seem from the PR. Changes between the already released version (9.0.4) and the current changeset are in deprecation of the three fields and three methods.
I'm not sure whether it's worth the new class and deprecate the old one.

Reply:
> That's not so easy... This map is not used just for ByUuid resolver, but for any work with the product in FE API (promoted product, lists), so the name is not ideal.

Makes sence, did not realize it before. Thanks!